src/strategy-factories/call-tokenuri/extractor.mjs

```diff
@@ -66,7 +70,7 @@ export const callTokenUriFactory = (props) => {
           to: address,
           data,
         },
-        toHex(BLOCK_NUMBER),
+        toHex(Math.max(parseInt(blockNumber), BLOCK_NUMBER)),
```

Comment:
this change has nothing to do with block logs or filtering it must not be included in this pr. I'm stopping review here and close